### PR TITLE
Test enhancement

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Using the assertion APIs to assert the correct assertions.
- Using the `DataProvider` to collect the test cases.
- Adding the `expectedExceptionMessage` to assert the expected exception message.
- Add the Travis CI build. Please look at that the [Travis CI build log](https://travis-ci.org/peter279k/workday-planner/builds/415484949).

---
This change is [Reviewable](https://reviewable.io/reviews/phpexpertsinc/workday-planner/5)

### CODE REVIEW

File: tests/HolidayDetectorTest.php [hunk 1]
```diff
@@ -130,13 +150,10 @@ public function testShowsErrorForInvalidData()
         $invalidFile = realpath(__DIR__.'/../data/holidays') . '/invalid.json';
         file_put_contents($invalidFile, 'invalid JSON');
 
-        try {
-            new HolidayDetector('invalid');
-            $this->fail('Did not throw an error with invalid JSON.');
-        } catch (\RuntimeException $e) {
-            unlink($invalidFile);
```
Comment (hopeseekr) [inline tests/HolidayDetectorTest.php]: It does not seem like your version removes the invalid data file.

**Review by hopeseekr — CHANGES_REQUESTED** [review]
This is great! THANK YOU SO MUCH!

File: tests/HolidayDetectorTest.php [hunk 2]
```diff
@@ -130,13 +150,10 @@ public function testShowsErrorForInvalidData()
...
-            $this->assertEquals("Invalid holiday data for 'invalid'.", $e->getMessage());
-        }
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage("Invalid holiday data for 'invalid'.");
```
Comment (peter279k) [inline tests/HolidayDetectorTest.php]: I replace the `try...catch` expected exception approach with the `expectException` approach.
And I think this should be better than previous approach.

**Review by hopeseekr — APPROVED** [review]

**Comment by hopeseekr** [comment]
@peter279k I greatly apologize for approving the PR way way back on Halloween 2018 yet not hitting the Merge button until 4 1/2 months later.

Thank you **so** much!! I really appreciate your contribution!

**Comment by peter279k** [comment]
Hi @hopeseekr, thank you for your reply.

Never mind. I hope you concern this and merge this PR finally :).

**Pull request merged**